This change doesn't compile for me:
```
error[E0277]: the trait bound `keyframe::Keyframe<T>: Clone` is not satisfied
  --> src/sequence.rs:20:2
   |
18 | #[derive(Clone, Default)]
   |          ----- in this derive macro expansion
19 | pub struct AnimationSequence<T> {
20 |     pub(crate) sequence: Vec<Keyframe<T>>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `keyframe::Keyframe<T>`
   |
   = note: required because of the requirements on the impl of `Clone` for `Vec<keyframe::Keyframe<T>>`
   = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)
```

This problem is that `Keyframe` doesn't implement `Clone`, and it can't because it doesn't require that the contained value are clonable. 

Values need to be clonable to be able to tween between them, but they don't need to be clonable to be able to put them in an animation sequence. You can still extract a pair of discrete values out of the sequence even if the values aren't clonable (with `AnimationSequence::pair()`). 

You could implement `Clone` manually for any keyframes where T is clonable. After that a `Clone` implementation can be made for animation sequences.

--- a/src/keyframe.rs
+++ b/src/keyframe.rs
@@ -1,4 +1,4 @@
-use alloc::{boxed::Box, fmt};
+use alloc::{boxed::Box, fmt, sync::Arc};
 use num_traits::Float;
 
 use crate::{
@@ -8,10 +8,11 @@ use crate::{
 };
 
 /// Intermediate step in an animation sequence
+#[derive(Clone)]
 pub struct Keyframe<T> {
 	value: T,
 	pub(crate) time: f64,
-	function: Box<dyn EasingFunction + Send + Sync>,
+	function: Arc<dyn EasingFunction + Send + Sync>,
 }
 
 impl<T> Keyframe<T> {
@@ -27,7 +28,7 @@ impl<T> Keyframe<T> {
 		Keyframe::<T> {
 			value,
 			time: if time < F::zero() { 0.0 } else { as_f64(time) },
-			function: Box::new(function),
+			function: Arc::new(function),
 		}
 	}
 
@@ -42,7 +43,7 @@ impl<T> Keyframe<T> {
 		Keyframe::<T> {
 			value,
 			time: if time < F::zero() { 0.0 } else { as_f64(time) },
-			function,
+			function: function.into(),
 		}
 	}
 

